### PR TITLE
Project 3: api

### PR DESCRIPTION
**Pull request by SamanAtashi**
# Project 3: API
Features introduced are as follows:
- connect API to react/redux app

### CODE REVIEW

File: src/components/Books/Book/AddBook.jsx [hunk 1]
```diff
@@ -1,18 +1,22 @@
+/* eslint-disable camelcase */
 import React from 'react';
```
Comment (Emmyn5600) [inline src/components/Books/Book/AddBook.jsx]: - [x] Fix eslint errors without disabling the line of code

File: src/components/Books/Book/Book.jsx [hunk 2]
```diff
@@ -1,20 +1,40 @@
-import React from 'react';
+/* eslint-disable max-len */
+import React, { useEffect } from 'react';
 import { useSelector, useDispatch } from 'react-redux';
```
Comment (Emmyn5600) [inline src/components/Books/Book/Book.jsx]: - [x] Fix eslint errors without disabling the line of code

File: src/components/Books/Book/Book.jsx [hunk 3]
```diff
@@ -1,20 +1,40 @@
...
+
+    book.forEach((book) => {
+      dispatch(addBook(book));
+    });
+  };
+
+  useEffect(() => {
+    retrieveFromApi();
+  }, []);
+
```
Comment (Emmyn5600) [inline src/components/Books/Book/Book.jsx]: - [x] Let us validate our add button to prevent us to adding the empty data
cpp13

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUESTED :recycle:

Hi, @SamanAtashi 

Great work on implementing the project requirements, however there are some issues to be solved before you get approved but you are almost there:muscle:. 
   
 - The repo looks professional.:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
-  Git/GitHub flow best practices have been followed.:heavy_check_mark:
- Data are being sent and received from API:heavy_check_mark:

Changes required :writing_hand: 
 
    - Please check the comment under the review.

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: src/API/api_connect.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        item_id: id,
+        title,
+        category: author,
+      },
+    ),
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  });
+  console.log('sended to api');
```
Comment (oxenprogrammer) [inline src/API/api_connect.js:17]: **Optional Changes**
The previous Code Reviewer missed it, however;
- [ ] You shouldn't commit `console.log` to production. It makes your code look messy (code smell). 
Let us try to remove all the occurrences of `console.log` before merging.

File: src/components/Books/Book/AddBook.jsx [hunk 5]
```diff
@@ -2,18 +2,24 @@ import React from 'react';
...
       author: document.getElementById('author').value,
     };
-    // dispatch an action and pass it the newBook object (your action's payload)
-    dispatch(addBook(newBook));
+
+    if (document.getElementById('title').value !== '' && document.getElementById('author').value !== '') {
+      await StoreApiSend(newBook.title, newBook.author, newBook.id);
+      dispatch(addBook(newBook));
+    } else {
+      alert('Please fill both inputs!!!');
```
Comment (oxenprogrammer) [inline src/components/Books/Book/AddBook.jsx:21]: **Optional Changes**
The previous Code Reviewer missed it, however;
- [ ] Let us restrain from using `windows.alert` . Simply put, it is difficult to test. 
Let us try to remove it before merging.

**Review by oxenprogrammer — APPROVED** [review]
Hi **@SamanAtashi** , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**